### DISCUSSION TITLE
current output out of sync with current input

### DISCUSSION THREAD

**Discussion by swhan22** [pybamm-team/PyBaMM#2651] [Q&A] [answered]
I know people have asked similar questions before but I just cannot figure out why my current input is out of sync with current output. I am trying to simulate a drive cycle, and I have the following piece of code:

timescale = para.evaluate(DFNmodel.timescale)
para["Current function [A]"] = pb.Interpolant(drive_cycle[:, 0], drive_cycle[:, 1], timescale * pb.t) 
solver = pb.CasadiSolver(mode = 'safe', dt_max = 1e-6, root_tol = 1e-6, atol = 1e-6, rtol = 1e-6)
sim = pb.Simulation(DFNmodel, parameter_values = para, solver = solver)
t_eval = np.arange(0, 500, 0.1)
sol = sim.solve(t_eval)

When I compare between the current input and the current output profile from the solution, they are always out of sync by about 5-10 seconds. Could somebody please help me with this issue? Thanks.

<img width="396" alt="Screenshot 2023-02-02 at 4 05 39 PM" src="https://user-images.githubusercontent.com/119541904/216478902-2aaba95a-fe44-4a92-8c82-7c51ee04fef1.png">

**Comment by brosaplanella** [comment, marked as answer]
Can you include a full minimum working example (including what you plot)?

**Reply by swhan22** [reply]
I found what was causing the issue. I was tinkering with electrode height after I evaluated timescale (e.g., timescale = para.evaluate(model.timescale)). But I found that time scaling factor depends on electrode height as follows:

F_const * Maximum concentration in negative electrode [mol.m-3] * (Negative electrode thickness [m] + Separator thickness [m] + Positive electrode thickness [m]) / absolute(Typical current [A] / (Number of electrodes connected in parallel to make a cell * Electrode width [m] * Electrode height [m])).

I should have fixated above parameters before I evaluated timescale. Nonetheless, I am so grateful that you offered me help!
